pkg/build/builder/docker.go

```diff
@@ -64,16 +60,10 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from build config %s ...", d.build.Name)
-	var push bool
-
 	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
+	// (which requires a tag) will still work, but we won't push it at the end.
 	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
-		d.build.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: noOutputDefaultTag,
-		}
-		push = false
+		d.build.Status.OutputDockerImageReference = d.build.Name
```

Comment:
Before the existence of `d.build.Status.OutputDockerImageReference`, we used to look at `d.build.Spec.Output.To.Name` and check if it is the empty string. In that case, we'd fake the output reference, and disable push.

I don't understand why here we still check the emptiness of `d.build.Spec.Output.To.Name`, but however set `d.build.Status.OutputDockerImageReference` to `d.build.Name`?!

This looks wrong, as `d.build.Name` is something like "my-build-1", which has nothing to do with a OutputDockerImageReference...

@bparees @PI-Victor any thoughts? Maybe I'm missing the point.